src/System.Runtime.Extensions/src/System/IO/Path.Unix.cs

```diff
@@ -69,15 +147,38 @@ private static string RemoveLongPathPrefix(string path)
 
         public static string GetTempPath()
         {
-            // TODO: Implement this
-            throw new NotImplementedException();
+            const string TempEnvVar = "TMPDIR";
+            const string DefaultTempPath = "/tmp/";
+
+            // Get the temp path from the TMPDIR environment variable.
+            // If it's not set, just return the default path.
+            // If it is, return it, ensuring it ends with a slash.
+            string path = Environment.GetEnvironmentVariable(TempEnvVar);
+            return
+                string.IsNullOrEmpty(path) ? DefaultTempPath :
+                IsDirectorySeparator(path[path.Length - 1]) ? path :
+                path + DirectorySeparatorChar;
         }
 
         private static string InternalGetTempFileName(bool checkHost)
         {
-            // TODO: Implement this
-            throw new NotImplementedException();
-        }s
+            const string Suffix = ".tmp";
+            const int SuffixByteLength = 4;
+
+            // mkstemps takes a char* and overwrites the XXXXXX with six characters
+            // that'll result in a unique file name.
+            string template = GetTempPath() + "tmpXXXXXX" + Suffix + "\0";
+            byte[] name = Encoding.UTF8.GetBytes(template);
```

Comment:
Ah, right.  It's based on whatever the user's LANG value is set to; which looks to be en_US.UTF-8 by default on Ubuntu and whatever account creation I did on OSX.

Are we concerned about Unix situations where someone has their LANG set to C or POSIX (both are ASCII)? Perhaps we need an interop helper somewhere for getting the right encoding depending on our launch locale (preferably by using locale capability detection, not reading those environment variables)?  Might depend on the work @ellismg is doing... or maybe we just document that we assume everyone is UTF-8 these days.